src/executor_heredoc.c

```diff
@@ -89,24 +89,26 @@ t_err	ft_create_heredoc(t_cmd *cmd, char *delim, int curr_delim,
  * @param name		Name of heredoc.
  * @return t_err	ERR_ABORT, ERR_HEREDOC_EOF, SUCCESS
  */
-t_err	ft_read_heredoc(char *delim, char *prompt2, int fd, char **name)
+t_err	ft_read_heredoc(t_hdoc *heredoc, t_hashtable *symtab, char *prompt2)
 {
 	char	*buf;
-	size_t	len;
+	t_err	err;
 
 	ft_signal_setup(SIGINT, SIG_HEREDOC);
-	len = ft_strlen(delim);
 	while (1)
 	{
 		buf = readline(prompt2);
 		if (g_status == 130)
-			return (ft_unlink_heredoc(name, ERR_ABORT));
+			return (ERR_ABORT);
 		if (!buf)
-			return (ft_print_warning(delim));
-		if (ft_strncmp(delim, buf, len + 1) == 0)
+			return (ft_print_warning(heredoc->delim));
+		if (ft_strncmp(buf, heredoc->delim, heredoc->delim_len + 1) == 0)
+			break ;
+		if (heredoc->quoted == false)
+			err = ft_expander_heredoc(&buf, symtab);
+		if (err != SUCCESS)
 			break ;
-		write(fd, buf, ft_strlen(buf));
-		write(fd, "\n", 1); 
+		ft_putendl_fd(buf, heredoc->fd);
 		free(buf);
 		buf = NULL;
 	}
```

Comment:
return err, if ERR_MALLOC was encountered during expanding

Reply:
Added. Exact error control flow will be handled separately